frontend/src/redux/roadmaps/reducers.ts

```diff
@@ -165,43 +165,43 @@ export const DELETE_TASK_FULFILLED = (
   parent.tasks = parent.tasks.filter((task) => task.id !== action.payload.id);
 };
 
-export const ADD_TASKRATING_FULFILLED = (
+export const ADD_TASKRATINGS_FULFILLED = (
   state: RoadmapsState,
-  action: PayloadAction<Taskrating>,
+  action: PayloadAction<Taskrating[]>,
 ) => {
-  let parentTask: Task | undefined;
   if (!state.roadmaps) throw new Error('Roadmaps havent been fetched yet');
-  state.roadmaps.forEach((roadmap) => {
-    if (parentTask !== undefined) return;
-    parentTask = roadmap.tasks.find(
-      (task) => task.id === action.payload.parentTask,
-    );
-  });
 
-  if (parentTask) {
-    parentTask.ratings.push(action.payload);
-  }
+  action.payload.forEach((taskRating) => {
+    let parentTask: Task | undefined;
+    state.roadmaps!.forEach((roadmap) => {
+      if (parentTask !== undefined) return;
+      parentTask = roadmap.tasks.find(
+        (task) => task.id === taskRating.parentTask,
+      );
+    });
```

Comment:
This is repeated three times, and sufficiently non-trivial to benefit from extracting out. For example:

```ts
const findTask = (taskId: number, roadmaps: Roadmap[]) =>
  roadmaps.reduce<Task | undefined>(
    (found, roadmap) =>
      found ?? roadmap.tasks.find((task) => task.id === taskId),
    undefined,
  );
```

```suggestion
    const parentTask = findTask(taskRating.parentTask, state.roadmaps!);
```